pypesto/C.py

```diff
@@ -0,0 +1,127 @@
+"""
+Constants
+=========
+Package-wide consistent constant definitions.
+"""
+
+from enum import Enum
+
+from typing import Tuple, Union
+
+###############################################################################
+# OBJECTIVE
+
+MODE_FUN = 'mode_fun'  # mode for function values
```

Comment:
Might be worth reformatting this way:

```
class ObjectiveMode(enum.Enum, str):
    MODE_FUN = 'mode_fun' # function values
    MODE_RES = 'mode_res' # residuals
```

makes import easier, better input checking, etc.
should also address some of the issues pointed out below.

Reply:
would mess with a couple more files here, but sound good for another pr